contracts/mainnet/connectors/morpho-blue/main.sol

```diff
@@ -0,0 +1,918 @@
+				TokenInterface(wethAddr),
+				_amt
+			);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogWithdrawCollateralOnBehalf((address,address,address,address,unit256),uint256,address,address,uint256,uint256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_amt,
+			_onBehalf,
+			_receiver,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @notice Handles the withdrawal of supply.
+	 * @dev  The market to withdraw assets from. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _marketParams The market to withdraw assets from.
+	 * @param _assets The amount of assets to withdraw. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function withdraw(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _assets);
+
+		_marketParams = updateTokenAddresses(_marketParams);
+
+		if (_amt == type(uint256).max) {
+			bytes32 _id = id(_marketParams);
+			Position memory _pos = MORPHO_BLUE.position(_id, address(this));
+			uint256 _shares = _pos.supplyShares;
+
+			_amt = _toAssetsUp(
+				_shares,
+				MORPHO_BLUE.market(_id).totalSupplyAssets,
+				MORPHO_BLUE.market(_id).totalSupplyShares
+			);
+		}
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.withdraw(
+			_marketParams,
+			_amt,
+			0,
+			address(this),
+			address(this)
+		);
+
+		convertWethToEth(
+			_marketParams.loanToken == ethAddr,
+			TokenInterface(wethAddr),
+			_assets
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogWithdraw((address,address,address,address,unit256),uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @notice Handles the withdrawal of a specified amount of assets by a user from a specific market.
+	 * @dev The market to withdraw assets from. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _marketParams The parameters of the market.
+	 * @param _assets The amount of assets the user is withdrawing. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address who's position to withdraw from.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function withdrawOnBehalf(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		address _onBehalf,
+		address _receiver,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _assets);
+
+		_marketParams = updateTokenAddresses(_marketParams);
+
+		if (_amt == type(uint256).max) {
+			bytes32 _id = id(_marketParams);
+			Position memory _pos = MORPHO_BLUE.position(_id, _onBehalf);
+			uint256 _shares = _pos.supplyShares;
+
+			_amt = _toAssetsUp(
```

Comment:
same comment as for `withdraw`

Reply:
Updated [here](https://github.com/Instadapp/dsa-connectors-2.0/pull/1/files#diff-920626a6b7a6adcbdc1062029932b5a6d9b5ee7f2a461452bc13e62a5b7274b1R500-R515)